scripts/fuse/elektra_fuse/rootlevel_resolver.py

```diff
@@ -0,0 +1,286 @@
+        self.mountpoint = mountpoint
+        self.fd = 0
+
+    #e.g '/' and '/12' are considered rootlevel
+    @staticmethod
+    def is_rootlevel_path(path):
+        return len(Path(path).parts) <= 2
+
+    # resolves a path "/pid/suffix" to "/suffix", where pid need to be a valid process id
+    # returns the process imformation with the suffix 
+    def resolve_proc_path(self, proc_path):
+        matches = re.match("^/(\d+)(.*)$", proc_path)
+        if matches:
+            pid = int(matches.group(1))
+            path_suffix = matches.group(2)
+            if path_suffix == '':
+                path_suffix = "/"
+            
+            try:
+                process_context = mock_context.parse_process_context_of_pid(pid)
+
+                if self._is_path_child_of(process_context.cwd, self.mountpoint): #same rationale as self.readdir
+                    raise FileNotFoundError
+                
+                return process_context, path_suffix
+            except FileNotFoundError:
+                raise OSError(errno.ENOENT)
+        else:
+            raise OSError(errno.ENOENT)
+
+
+    def _exists_in_rootlevel(self, path):
+        return path == "/" or Path(path).name in self._get_all_pids()
+        #TODO: cannot use self.readdir here, as this triggers an endless recusion. Meanwhile, excluded pids are falsely flagged as existing.
+        # (which only leads to unexpected behaviour in case of especially crafted requests)
+
+    
+    @with_translated_exceptions
+    def getattr(self, path, fh=None):
+
+        generic_dir_attrs = dict(
+            st_mode = (stat.S_IFDIR | 0o755),
+            st_ctime = elektra_fuse_interface.startup_time,
+            st_mtime = elektra_fuse_interface.startup_time,
+            st_atime = elektra_fuse_interface.startup_time,
+            st_nlink = 2)
+
+        if RootlevelResolver.is_rootlevel_path(path):
+            if self._exists_in_rootlevel(path):
+                return generic_dir_attrs
+            else:
+                raise OSError(errno.ENOENT)
+
+        process_context, path_suffix = self.resolve_proc_path(path)
+
+        return mock_context.run_as(process_context, elektra_fuse_interface.getattr, path_suffix, fh=fh)
+
+    def _get_all_pids(self):
+        return {file for file in os.listdir("/proc") if file.isnumeric()}
+
+    def _is_path_child_of(self, potential_child_path, potential_ancestor_path):
+        potential_child_path = Path(potential_child_path).resolve()
+        potential_ancestor_path = Path(potential_ancestor_path).resolve()
+        return potential_ancestor_path in [*potential_child_path.parents, potential_child_path]
+
+    @with_translated_exceptions
+    def readdir(self, path, fh):
+
+        if path == "/":
+            all_pids = self._get_all_pids()
+
+            #filter out processes with working dir under mountpoint to prevent endless recursions (e.g. an interactive shell navigatung below the mountpoint)
```

Comment:
```suggestion
            #filter out processes with working directory under mount point to prevent endless recursions (e.g. an interactive shell navigating below the mount point)
```